kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -38,10 +39,13 @@ import scala.util.{Failure, Success}
   * [[KafkaConsumerConfig]] needed and see `monix/kafka/default.conf`,
   * (in the resource files) that is exposing all default values.
   */
-final class KafkaConsumerObservable[K, V] private (config: KafkaConsumerConfig, consumer: Task[KafkaConsumer[K, V]])
-    extends Observable[ConsumerRecord[K, V]] {
+final class KafkaConsumerObservable[K, V, Out] private (
+  config: KafkaConsumerConfig,
+  consumer: Task[KafkaConsumer[K, V]],
+  transformOutput: (KafkaConsumer[K, V], Iterable[ConsumerRecord[K, V]]) => Iterable[Out])
```

Comment:
Maybe we could just create base trait to share common logic and have two separate classes? 
There is some code and configuration that only concerns one of the option and not the other + signature should be nicer

Reply:
Yes, It is the most doubtful place in the PR. 

What do you think about this snippet?

```scala
trait KafkaConsumerObservable[K, V, Out] extends Observable[Out] {
   protected def config: KafkaConsumerConfig
   protected def consumer: Task[KafkaConsumer[K, V]]
   def transformOutput(consumer: KafkaConsumer[K, V], out: Iterable[ConsumerRecord[K, V]]): Iterable[Out]

   def unsafeSubscribeFn(out: Subscriber[Out]): Cancelable = {
    import out.scheduler

    val callback = new Callback[Throwable, Unit] {
      def onSuccess(value: Unit): Unit =
        out.onComplete()
      def onError(ex: Throwable): Unit =
        out.onError(ex)
    }

    feedTask(out).runAsync(callback)
  }

  private def feedTask(out: Subscriber[Out]): Task[Unit] = {
    // ...
  }
}

final class KafkaConsumerObservableAutoCommit[K, V] private (
  override val config: KafkaConsumerConfig,
  override val consumer: Task[KafkaConsumer[K, V]]) extends KafkaConsumerObservable[K, V, ConsumerRecord[K, V]] {
  override def transformOutput(consumer: KafkaConsumer[K, V], out: Iterable[ConsumerRecord[K, V]]): Iterable[Out] = identity
}

final class KafkaConsumerObservableManualCommit[K, V] private (
  override val config: KafkaConsumerConfig,
  override val consumer: Task[KafkaConsumer[K, V]]) extends KafkaConsumerObservable[K, V, CommittableMessage[K, V]] {
  override def transformOutput(consumer: KafkaConsumer[K, V], out: Iterable[ConsumerRecord[K, V]]): Iterable[Out] = ???
}

```

Reply:
Ideally we could avoid having `transformOutput` function in standard version and get rid of few `if`s in manual commit

Meaning we could share pretty much everything apart from up to 50 lines in `runLoop`

Reply:
I did it. Take a look, please.